applicationset/services/repo_service.go

```diff
@@ -54,6 +58,7 @@ func (a *argoCDService) GetFiles(ctx context.Context, repoURL string, revision s
 		return nil, err
 	}
 
+	defer a.keyLock.Lock(repoURL)()
```

Comment:
Locking on the repo URL doesnt help too much if you have a monorepo where the config files for your app live. It would make more sense to lock based on something like `<app set name>/<repo URL>/`